### ISSUE TITLE
Programmably Disable Features

### ISSUE THREAD

**Issue by mlucool** [zalmoxisus/redux-devtools-extension#289]
In reference to https://github.com/zalmoxisus/redux-devtools-extension/issues/185, there are times I'd like to be able to explore the data visually with this in production, but for safety reasons disable any feature which might change the state from this tool including:

- Importing previous state
- Replaying state in any means (i.e. time traveling , removing an event by clicking on it)

I want to avoid actions not coming from the UI to prevent non idempotent services from being called etc.

**Comment by zalmoxisus** [comment]
If it's for security reason, the issue here is that even if we restrict this from the extension part, a malefic script can hijack the extension's global variable, and use its store enhancer instead. Of course, it's not that simple, and it should be executed before the store is created.

I was thinking to add a helper, which would include [`redux-devtools-instrument`](https://github.com/zalmoxisus/redux-devtools-instrument) outside of the extension, so it wouldn't have direct access to the store. However, if we exclude monitor actions, as you suggested, there's no reason to include this script. We could just add the logging part in a separate script you can include like so:
```js
- import { composeWithDevTools } from 'redux-devtools-extension';
+ import { composeWithDevTools } from 'redux-devtools-extension/logOnly';
```

**Comment by mlucool** [comment]
Yeah - the logger only would be great for prod. It's less about security and more about preventing mistakes.

**Comment by zalmoxisus** [comment]
I just published `redux-devtools-extension@2.0.0` to npm, which implements that. Use as above:
```js
import { composeWithDevTools } from 'redux-devtools-extension/logOnly'
```

You could even use it like so:
```js
const { composeWithDevTools } = require(
  process.env.NODE_ENV === 'production' ?
  'redux-devtools-extension/logOnly' : 'redux-devtools-extension'
)
```

Or with `import()` in Webpack 2.

The code is [pretty simple](https://github.com/zalmoxisus/redux-devtools-extension/blob/master/npm-package/logOnly.js) and will not add signifiant weight to the production bundle.

Will add support for special data (and for `immutable` parameter) tomorrow and will publish the extension as well. Also we need the ability to start / stop recording. Anyway, you can give it a try with the current extension without those features.

**Comment by mlucool** [comment]
I only use this extension via the chrome extension. Is this available there?

**Comment by zalmoxisus** [comment]
You still need the npm package. There will be a blog post with more details. [Here's a draft](https://medium.com/@zalmoxis/using-redux-devtools-in-production-4c5b56c5600f). Please review it and let me know if you think something can be improved. Also there are [lots of changes](https://github.com/zalmoxisus/redux-devtools-extension/commits/master), which I'll publish soon.

**Comment by mlucool** [comment]
This all seems great. My only concern is why do we need the npm package? It seems not ideal to have yet another dependency, especially when it needs to be aligned with the plugin. Can the plugin expose this package and I can do the logOnly webpacking myself?

**Comment by zalmoxisus** [comment]
The main purpose of that helper is not to expose the store globally, but via a local script. That was usually the concern of including the extension in production.

If that is not an issue for you, as mentioned in the article, you will be able to use the extension as before, and just specify which features you want to allow:
```js
const composeEnhancers =
  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
      features: {
        pause: true, // start/pause recording of dispatched actions
        lock: true, // lock/unlock dispatching actions and side effects    
        persist: true, // persist states on page reloading
        export: true, // export history of actions in a file
        import: 'custom', // import history of actions from a file
        jump: true, // jump back and forth (time travelling)
        reorder: true, // drag and drop actions in the history list 
        dispatch: true, // dispatch custom actions or action creators
        test: true // generate tests for the selected actions
      },
      // other options like actionSanitizer, stateSanitizer
})
```

**Comment by mlucool** [comment]
Yeah, that's not an issue for me at all. I just want to prevent people from shooting themselves in the foot. I would make it clearer this package is optional.  It read to me as using this package then restrict. Otherwise, looks great.

**Comment by zalmoxisus** [comment]
Great! I just published `2.13.0` on Chrome Store. By default all features are enabled, unless you specify `features` parameter. Parts like slider and dispatcher will be hidden if not enabled. The buttons inside the monitors are not hidden, just have no effect. We'll hide them in `3.0`.

**Comment by mlucool** [comment]
This is fantastic - thanks!
